app/bundles/LeadBundle/Form/Type/CompanyListType.php

```diff
@@ -0,0 +1,84 @@
+<?php
+/**
+ * @package     Mautic
+ * @copyright   2014 Mautic Contributors. All rights reserved.
+ * @author      Mautic
+ * @link        http://mautic.org
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+namespace Mautic\LeadBundle\Form\Type;
+
+use Mautic\CoreBundle\Factory\MauticFactory;
+use Mautic\LeadBundle\Entity\CompanyRepository;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\OptionsResolver\Options;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * Class CompanyListType
+ *
+ * @package Mautic\LeadBundle\Form\Type
+ */
+class CompanyListType extends AbstractType
+{
+    /**
+     * @var CompanyRepository
+     */
+    private $repo;
+
+    /**
+     * @var bool
+     */
+    private $viewOther;
+
+    /**
+     * @param MauticFactory $factory
+     */
+    public function __construct(MauticFactory $factory)
```

Comment:
Same as above. Let's not use MauticFactory is deprecated and shouldn't be used in new code. Inject the needed services as dependencies instead.